### PR TITLE
Support TypeScript 5.3

### LINKED ISSUE #15699: Support for import attributes with TypeScript
**Environments:**

- Prettier Version: 3.1.0
- Running Prettier via: VS Code plugin (`esbenp.prettier-vscode`), also via ESLint and `eslint-plugin-prettier@5.0.1`
- Runtime: Node.js v21.2.0
- Operating System: Dev Container based on Alpine Linux 3.18.4
- Prettier plugins (if any): –

**Steps to reproduce:**

Process the file

```typescript
import packageInfo from "../../package.json" with { type: "json" };
```

with TypeScript parser.

[https://prettier.io/playground/…](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgAkwEMwBrIgczgEkoAzCfOnCdfAHRADouB6H3sTKU4XAFYBnaJ3wB3VDAAW+YPhgBPTHCQcQk6SHwBfANzsoIADQgImGKmgTkoIjhayACq4ROURADayROpO1gBGOCSkcDAAykTocAAyqFBwyHQBEnDhkWQxsUKpFMgwOACuOSDZ6KilFVVwAB5aOBgIMAEAKnA4UK6ocL6Z-tnWEsX+cACK5RDwGVlVkk2xkzNzC0gjYyAAjptwHiyYviBEEgC0aXAAJndWIGVEqP7FAMKs6ETI5-7+jwmUAoUwAgjAyqgwuV4B5eik0otRlVFDB0P4AOqKBRDIRwWI+BSoABuCnUvzAElCIGJlVo91gsTAbTsoKgt1iGimSN2mBY2QxkUwvz5Q16xPS1lS2Twx0o3x5VWIOBlvw0WgkzNQdkefNSMAxqFuSmQAA4AAzWHBwA6oa1yigK7ZLaydMKG42KZAAJms5WyXSIYWGLpAcHQYTu91uSSIwPKIgAYrhvhDir8iDCICAjEYgA)

**Expected behavior:**

No errors

**Actual behavior:**

`SyntaxError: ';' expected. (1:46)`

### PR DESCRIPTION
**Pull request by fisker**
## Description

Fixes #15699

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/parse/postprocess/index.js [hunk 1]
```diff
@@ -111,8 +117,22 @@ function postprocess(ast, options) {
             };
           }
         }
+
+        // In TypeScript, we can't distinguish `with`/`assert`
```
Comment (sosukesuzuki) [inline src/language-js/parse/postprocess/index.js:121]: Oh...

Comment (sosukesuzuki) [reply src/language-js/parse/postprocess/index.js:121]: Sorry, I have not been able to research this yet. If you know, please let me know.
Is this a problem with TypeScript Compiler? Or is it a typescript-estree issue? If it is a typescript-estree issue, is there any way to fix the parser side?

Comment (fisker) [reply src/language-js/parse/postprocess/index.js:121]: I don't think typescript-estree cares about it, I think typescript treat them the same.

**Comment by ForsakenHarmony** [comment]
Is there any chance this will be released as a `3.1.1` or `3.2` (or `4.0`) soon?

**Pull request merged**